catroid/src/org/catrobat/catroid/ui/dialogs/UploadProjectDialog.java

```diff
@@ -265,6 +265,12 @@ private void handleUploadButtonClick() {
 	}
 
 	private void handleCancelButtonClick() {
+		Context context = getActivity().getApplicationContext();
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		if (sharedPreferences.getBoolean(Constants.RESTRICTED_USER, false)) {
+			sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).commit();
+			sharedPreferences.edit().putString(Constants.USERNAME, Constants.NO_USERNAME).commit();
+		}
```

Comment:
Duplicate code
Consider to use a method and use it in ProjectUploadService.java as well.